core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -88,7 +112,17 @@ public T acquire() {
           if (resource != null) {
             return resource;
           }
-          mNotEmpty.await();
+          if (time != null) {
+            long currTimeMs = System.currentTimeMillis();
```

Comment:
Nevermind, the contract can be time spent within the method instead of waiting for the lock.